src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -594,11 +603,119 @@ def get_household_structure(pop: pd.DataFrame) -> pd.DataFrame:
         return household_structure
 
     @staticmethod
-    def choose_random_guardian(member_ids: pd.DataFrame) -> pd.Series:
+    def choose_random_guardian(member_ids: pd.DataFrame, groupby_level: str) -> pd.Series:
         # member_ids is a subset of child_households dataframe
+        # groupby_level will be index level to group by (the first level index of member_ids (child_id or mother_id).
         member_ids = (
             member_ids.reset_index()
-            .groupby(["child_id"])["person_id"]
+            .groupby([groupby_level])["person_id"]
             .apply(np.random.choice)
         )
         return member_ids
+
+    def assign_second_guardian_to_newborns(
+        self, new_births: pd.DataFrame, households: pd.DataFrame
+    ) -> pd.DataFrame:
+        """
+        Parameters
+        ----------
+        new_births: pd.DataFrame of new births on this time step
+        households: pd.DataFrame of the entire state table for the two columns of household_id
+          and relation_to_household_head.
+
+        Returns
+        -------
+        new_births with the additional guardian_2 column updated.
+        """
+        # Setup
+        new_births["guardian_2"] = data_values.UNKNOWN_GUARDIAN_IDX
+        mothers_households = self.get_mothers_household_structure(
+            new_births["parent_id"], households
+        )
+        partners = [
+            "Opp-sex spouse",
+            "Opp-sex partner",
+            "Same-sex spouse",
+            "Same-sex partner",
+        ]
+
+        # Index helpers
+        # Mother index groups
+        mother_rp_idx = mothers_households.loc[
+            mothers_households["mother_relation_to_household_head"] == "Reference person"
+        ].index
+        mother_partner_idx = mothers_households.loc[
+            mothers_households["mother_relation_to_household_head"].isin(partners)
+        ].index
+        # Potential partner index groups
+        partners_idx = mothers_households.loc[
+            mothers_households["member_relation_to_household_head"].isin(partners)
+        ].index
+        reference_person_idx = mothers_households.loc[
+            mothers_households["member_relation_to_household_head"] == "Reference person"
+        ].index
+
+        # Assign second guardian to random partner of mothers
+        partner_ids = mothers_households.loc[mother_rp_idx.intersection(partners_idx)]
+        if len(partner_ids) > 0:
+            partner_ids = self.choose_random_guardian(partner_ids, "mother_id")
```

Comment:
This needs a method name change if it's now used for something other than choosing random guardians

Reply:
Oh I'm confused then. I thought this was choosing a random partner for the female which then may or may not be assigned as guardian of the child.

Reply:
choose_random_guardian is picking a random guardian from a subset of mother_households here (or in the other use cases child_households).  This is why there is always a preceding check for len since you can't groupby on an object of length 0.